pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -3266,6 +3266,68 @@ func TestChangefeedRetryableError(t *testing.T) {
 	t.Run(`pubsub`, pubsubTest(testFn))
 }
 
+func TestChangefeedJobRetryOnNoInboundStream(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	cluster, db, cleanup := startTestCluster(t)
+	defer cleanup()
+	sqlDB := sqlutils.MakeSQLRunner(db)
+
+	// force fast "no inbound stream" error
+	var oldMaxRunningFlows int
+	var oldTimeout string
+	sqlDB.QueryRow(t, "SHOW CLUSTER SETTING sql.distsql.max_running_flows").Scan(&oldMaxRunningFlows)
+	sqlDB.QueryRow(t, "SHOW CLUSTER SETTING sql.distsql.flow_stream_timeout").Scan(&oldTimeout)
+	sqlDB.Exec(t, "SET CLUSTER SETTING sql.distsql.max_running_flows=0")
+	sqlDB.Exec(t, "SET CLUSTER SETTING sql.distsql.flow_stream_timeout='1s'")
+	testutils.SucceedsSoon(t, func() error {
+		for nodeID := 0; nodeID < cluster.NumServers(); nodeID++ {
+			var flows int
+			conn := cluster.ServerConn(nodeID)
+			db := sqlutils.MakeSQLRunner(conn)
+			db.QueryRow(t, "SHOW CLUSTER SETTING sql.distsql.max_running_flows").Scan(&flows)
+			if flows > 0 {
+				return errors.New("old max_running_flows value")
+			}
+		}
```

Comment:
Have you seen `serverutils.SetClusterSetting`? I added that specifically because of propagation issue you're trying to work around.